python_modules/libraries/dagster-airbyte/dagster_airbyte/asset_defs.py

```diff
@@ -88,3 +93,209 @@ def _assets(context):
+            out += _get_normalization_tables_for_schema(k, v, f"{prefix}{key}_")
+    return out
+
+
+def _clean_name(name: str) -> str:
+    """
+    Cleans an input to be a valid Dagster asset name.
+    """
+    return "".join(c if VALID_NAME_REGEX.match(c) else "_" for c in name)
+
+
+class AirbyteConnection(
+    NamedTuple(
+        "_AirbyteConnection",
+        [
+            ("name", str),
+            ("source_config_path", str),
+            ("stream_prefix", str),
+            ("has_basic_normalization", bool),
+            ("stream_data", List[Mapping[str, Any]]),
+        ],
+    )
+):
+    """
+    Contains information about an Airbyte connection.
+
+    Attributes:
+        name (str): The name of the connection.
+        source_config_path (str): The path to the Airbyte source configuration file.
+        stream_prefix (str): A prefix to add to all stream names.
+        has_basic_normalization (bool): Whether or not the connection has basic normalization enabled.
+        stream_data (List[Mapping[str, Any]]): Unparsed list of dicts with information about each stream.
+    """
+
+    def parse_stream_tables(
+        self, return_normalization_tables: bool = False
+    ) -> Mapping[str, Set[str]]:
+        """
+        Parses the stream data and returns a mapping, with keys representing destination
+        tables associated with each enabled stream and values representing any affiliated
+        tables created by Airbyte's normalization process, if enabled.
+        """
+
+        tables: Dict[str, Set[str]] = {}
+
+        enabled_streams = [
+            stream for stream in self.stream_data if stream.get("config", {}).get("selected", False)
+        ]
+
+        for stream in enabled_streams:
+            name = cast(str, stream.get("stream", {}).get("name"))
+            prefixed_name = f"{self.stream_prefix}{name}"
+
+            tables[prefixed_name] = set()
+            if self.has_basic_normalization and return_normalization_tables:
+                for k, v in stream["stream"]["json_schema"]["properties"].items():
+                    for normalization_table_name in _get_normalization_tables_for_schema(
+                        k, v, f"{name}_"
+                    ):
+                        prefixed_norm_table_name = f"{self.stream_prefix}{normalization_table_name}"
+                        tables[prefixed_name].add(prefixed_norm_table_name)
+
+        return tables
+
+
+class AirbyteSource(NamedTuple("_AirbyteSource", [("name", str)])):
+    """
+    Contains information about an Airbyte source.
+
+    Attributes:
+        name (str): The name of the source.
+    """
+
+
+def _airbyte_connection_from_config(contents: Mapping[str, Any]) -> AirbyteConnection:
+    config_contents = cast(Mapping[str, Any], contents.get("configuration"))
+    check.invariant(
+        config_contents is not None, "Airbyte connection config is missing 'configuration' key"
+    )
+
+    return AirbyteConnection(
```

Comment:
I think it'd be marginally cleaner to have an `AirbyteConnection.from_config` method, rather than having this helper function. It co-locates the logic for how each of these fields is populated with the object itself.